Sources/Features/Breadcrumb/BacktraceBreadcrumbFileHelper.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+import Cassette
+
+enum BacktraceBreadcrumbFileHelperError: Error {
+    case invalidFormat
+}
+
+@objc public class BacktraceBreadcrumbFileHelper: NSObject {
+    
+    private var minimumQueueFileSizeBytes = 4096;
+    private var maxQueueFileSizeBytes = 0
+    private var breadcrumbLogDirectory: String = ""
+        
+    private var queue: QueueFile?
+    
+    private var queueByteSize: Int {
+        var size = 0
+        queue?.forEach({ data in
+            size += data?.count ?? 0
+            return true
+        })
+       return size
+    }
+    
+    public init(_ breadcrumbLogDirectory: String, maxQueueFileSizeBytes: Int) {
+        super.init()
+        self.queue = QueueFile.init(path: breadcrumbLogDirectory)
+        self.breadcrumbLogDirectory =  breadcrumbLogDirectory
+        if (maxQueueFileSizeBytes < minimumQueueFileSizeBytes) {
+            self.maxQueueFileSizeBytes = minimumQueueFileSizeBytes;
+        } else {
+            self.maxQueueFileSizeBytes = maxQueueFileSizeBytes;
+        }
+    }
+     
+    func addBreadcrumb(_ breadcrumb:[String:Any]) -> Bool {
+        do {
+            var text = try convertBreadcrumbIntoString(breadcrumb)
+            if queue?.isEmpty() == false {
+                text = "\n\n\(text)"
+            }
+            let textBytes = Data(text.utf8)
+            if textBytes.count > 4096 {
+                BacktraceLogger.error("We should not have a breadcrumb this big, this is a bug!")
```

Comment:
I disagree! how about a situation when I as a user want to store really long breadcrumb manually? Is this still an internal bug? I don't think so. In this situation also ... does it make sense to just cut the breadcrumb content instead of removing the whole breadcrumb?

Reply:
Fair point, however, this exact logic is present in Android as well. I will add the improvement to a ticket to improve both.